Upstream: under review, https://github.com/libgeos/geos/pull/805

From 0af19545b975f01f5212cd0a7f72e958c726d14a Mon Sep 17 00:00:00 2001
From: Heiko Becker <heirecka@exherbo.org>
Date: Sat, 21 Jan 2023 11:02:13 +0100
Subject: [PATCH] Fix build with gcc 13 by including <cstdint>

In addition to [1]. Otherwise geos fails to build with:
"geos-3.11.1/include/geos/shape/fractal/HilbertEncoder.h:40:28: error: expected ')' before 'p_level'
   40 |     HilbertEncoder(uint32_t p_level, geom::Envelope& extent);
      |                   ~        ^~~~~~~~
      |                            )
/var/tmp/paludis/build/sci-libs-geos-3.11.1/work/geos-3.11.1/include/geos/shape/fractal/HilbertEncoder.h:41:5: error: 'uint32_t' does not name a type
   41 |     uint32_t encode(const geom::Envelope* env);
      |     ^~~~~~~~"

[1] 0e8d4368b8bd72a7d361286e8523ebce5cff6146
---
 include/geos/shape/fractal/HilbertEncoder.h | 1 +
 tests/unit/capi/GEOSMakeValidTest.cpp       | 1 +
 2 files changed, 2 insertions(+)

diff --git a/include/geos/shape/fractal/HilbertEncoder.h b/include/geos/shape/fractal/HilbertEncoder.h
index 9e1c5fc22..9d270f9d2 100644
--- a/include/geos/shape/fractal/HilbertEncoder.h
+++ b/include/geos/shape/fractal/HilbertEncoder.h
@@ -19,6 +19,7 @@
 #include <geos/geom/Geometry.h>
 #include <string>
 #include <vector>
+#include <cstdint>
 
 // Forward declarations
 namespace geos {
diff --git a/tests/unit/capi/GEOSMakeValidTest.cpp b/tests/unit/capi/GEOSMakeValidTest.cpp
index 6a78ef7c9..7382fe7ca 100644
--- a/tests/unit/capi/GEOSMakeValidTest.cpp
+++ b/tests/unit/capi/GEOSMakeValidTest.cpp
@@ -9,6 +9,7 @@
 #include <cstdlib>
 #include <cmath>
 #include <cstring>
+#include <cstdint>
 
 #include "capi_test_utils.h"
 
-- 
2.39.1

